pkg/controllers/routercerts/controller.go

```diff
@@ -139,20 +139,30 @@ func (c *routerCertsDomainValidationController) validateRouterCertificates() ope
 	}
 
 	// get router certs secret
-	certBytes, _, isDefault, err := common.GetActiveRouterCertKeyBytes(c.secretLister, ingress, c.secretNamespace, c.defaultSecretName, c.customSecretName)
+	secret, err := common.GetActiveRouterSecret(c.secretLister, c.secretNamespace, c.defaultSecretName, c.customSecretName)
 	if err != nil {
 		return newRouterCertsDegradedf("NoRouterCertSecret", "neither the custom secret/%v -n %v or default secret/%v -n %v could be retrieved: %v", c.defaultSecretName, c.secretNamespace, c.customSecretName, c.secretNamespace, err)
 	}
 
+	// Perform a no-op if non-default secret is in use
+	if secret.GetName() != c.defaultSecretName {
+		return operatorv1.OperatorCondition{
+			Type:   conditionRouterCertsDegradedType,
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}
+	}
+
 	// cert data should exist
-	if len(certBytes) == 0 {
-		return newRouterCertsDegradedf("MissingRouterCertsPEM", "secret/%v -n %v: certificate not found", c.activeSecretName(isDefault), c.secretNamespace)
+	data := secret.Data[ingressDomain]
```

Comment:
nit: `s/data/certBytes/`

Reply:
I'm going to leave this as data because I reverted the earlier changes of this file based on the decision to perform a no-op when the default secret is not in use.